pkg/gatewayserver/gatewayserver.go

```diff
@@ -675,17 +674,17 @@ func (gs *GatewayServer) updateConnStats(conn connectionEntry) {
 		case <-ctx.Done():
 			return
 		case <-conn.StatsChanged():
-			err := gs.UpdateConnectionStats(ctx, conn.Connection)
-			if err != nil {
-				logger.WithError(err).Error("Failed to update connection stats")
-			}
+		}
 
-			timeout := time.After(gs.updateConnectionStatsDebounceTime)
-			select {
-			case <-ctx.Done():
-				return
-			case <-timeout:
-			}
+		if err := conn.UpdateStats(gs.statsRegistry); err != nil {
```

Comment:
Can you keep the timestamps here locally in this function, and pass to `UpdateStats()` what exactly needs to be updated?

Better yet, can we keep the `io` layer separate from persistency? Is there a reason to add storage to `io.Connection`?

Reply:
Moved logic into the go-routine instead.